Include/pymacro.h

```diff
@@ -14,7 +14,7 @@
 // compiler support (gcc >= 4.6) and is called _Static_assert.
 #if (defined(__GLIBC__) \
      && (__GLIBC__ < 2 || (__GLIBC__ == 2 && __GLIBC_MINOR__ <= 16)) \
```

Comment:
Ordering suggestion -- put all version checks first (had to split the suggestion due to the GH UI). I'm not sure what precedent here is though.
```suggestion
     && (__GLIBC__ < 2 || (__GLIBC__ == 2 && __GLIBC_MINOR__ <= 16)) \
     && !(defined(__cplusplus) && __cplusplus >= 201103L) \
```